lib/jekyll/hooks.rb

```diff
@@ -0,0 +1,103 @@
+module Jekyll
+  module Hooks
+    # Helps look up hooks from the registry by owner's class
+    OWNER_MAP = {
+      Jekyll::Site => :site,
+      Jekyll::Page => :page,
+      Jekyll::Post => :post,
+      Jekyll::Document => :document,
+    }.freeze
```

Comment:
Good idea. I think it's safe to proceed with this frozen constant for now because its just an internal API. If we need a public API for adding hook points we can change this storage mechanism easily enough.